python-avd/pyavd/_schema/avddataconverter.py

```diff
@@ -173,11 +175,31 @@ def deprecation(self, deprecation: dict, _data: Any, _schema: dict, path: list)
         if not deprecation.get("warning", True):
             return
 
-        yield AvdDeprecationWarning(
+        new_key = deprecation.get("new_key")
+        removed = deprecation.get("removed", False)
+
+        # If new_key set, we can check for collision where both new and old key are set.
```

Comment:
maybe this should make it to some page where we explain what we are able to catch and what we are not

Reply:
added in https://github.com/aristanetworks/avd/pull/4562/commits/c3f8e8d816b27b5ba7515a00c4ceefb14db8751a